singer_sdk/helpers/_typing.py

```diff
@@ -489,6 +506,9 @@ def _conform_primitive_property(  # noqa: PLR0911
         if is_boolean_type(property_schema):
             bit_representation = elem != b"\x00"
             return bit_representation
+        # if property_schema.get("contentEncoding"):
+        if is_encoded_string_type(property_schema):
+            return elem
```

Comment:
Should this still be encoded as a string? I'm not sure how JSON libraries deal with bytes values.

Reply:
I believe they should based on documentation if found at json-schema.org.  Please give it a look let me know what you think.

> JSON schema has a set of keywords to describe and optionally validate non-JSON data stored inside JSON strings. Since it would be difficult to write validators for many media types, JSON schema validators are not required to validate the contents of JSON strings based on these keywords. However, these keywords are still useful for an application that consumes validated JSON.

https://json-schema.org/understanding-json-schema/reference/non_json_data.html#media-string-encoding-non-json-data

Reply:
I agree that the values should be `str` before any JSON-encoding takes place.  Currently the '.hex()' default does this since it returns a string.  What if we ensure the values returned from the `bytes` portion of ` _conform_primitive_property` are valid ascii strings?  Since Batch messages don't go through this process and currently the `json.dumps` doesn't have a `default=str` we could extend the default encoding class to handle the byte type and pass that class and the encoder for it to use?  If you have that type of jsonencoder class would it make sense to use it for both record and batch message json.dumps?